src/api/app/services/staging/staging_project_creator.rb

```diff
@@ -1,17 +1,25 @@
 module Staging
   class StagingProjectCreator
-    def initialize(request_body, staging_workflow)
+    def initialize(request_body, staging_workflow, user)
       @request_body = request_body
       @staging_workflow = staging_workflow
+      @user = user
     end
 
     def call
       if @request_body.blank?
         errors << 'Staging projects are empty'
         return self
       end
-      projects = staging_projects
-      projects.each(&:store) if valid?
+
+      if valid?
+        projects = staging_projects
```

Comment:
@dmarcoux, this line needs to be outside the statement the `if`, otherwise it will be always `valid`.

Reply:
Oh, you are right :+1: